### PR TITLE
ssl/statem: Replace size_t with int and add the checks

### PR DESCRIPTION
**Pull request by JiangJias**
Replace the type of variables with int to avoid implicit conversion when it is assigned by EVP_MD_get_size().
Moreover, add the checks to avoid integer overflow.

Fixes: https://github.com/openssl/openssl/commit/1053a6e2281d81cd5d04d2d90da2c4905c9c3561 ("Implement Server side of PSK extension parsing")
Fixes: https://github.com/openssl/openssl/commit/6594189fa16e845df5565ca4c180220783a752d4 ("Merge early_data_info extension into early_data")
Fixes: https://github.com/openssl/openssl/commit/9368f86526d9feb1f511fab9c0eee8e8dc869b87 ("Add TLSv1.3 client side external PSK support")

##### Checklist

- [ ] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

File: ssl/statem/extensions_clnt.c [hunk 1]
```diff
@@ -952,8 +952,11 @@ EXT_RETURN tls_construct_ctos_padding(SSL_CONNECTION *s, WPACKET *pkt,
              * Add the fixed PSK overhead, the identity length and the binder
              * length.
              */
+            int tmpsize = EVP_MD_get_size(md);
+            if (tmpsize < 0)
```
Comment (t8m) [inline ssl/statem/extensions_clnt.c:956]: `<= 0` as 0 length is also not possible. And please add an empty line above.

**Review by t8m — CHANGES_REQUESTED** [review]

File: ssl/statem/extensions_clnt.c (hunk 1 above)
Comment (JiangJias) [reply ssl/statem/extensions_clnt.c]: Thanks. I have force-pushed a new commit to fix the problem.

Comment (t8m) [reply ssl/statem/extensions_clnt.c]: @JiangJias Could you please change similarly all the places where you check the md size?

Comment (JiangJias) [reply ssl/statem/extensions_clnt.c]: Fine

**Review by t8m — APPROVED** [review]

**Review by tom-cosgrove-arm — APPROVED** [review]
LGTM

**Comment by t8m** [comment]
Squashed and merged to the master branch. Thank you for your contribution.

**Pull request closed without merging**